yt_dlp/extractor/archiveorg.py

```diff
@@ -698,18 +891,21 @@ def _real_extract(self, url):
         url_date = url_date or url_date_2
 
         urlh = None
-        try:
-            urlh = self._request_webpage(
-                HEADRequest('https://web.archive.org/web/2oe_/http://wayback-fakeurl.archive.org/yt/%s' % video_id),
-                video_id, note='Fetching archived video file url', expected_status=True)
-        except ExtractorError as e:
-            # HTTP Error 404 is expected if the video is not saved.
-            if isinstance(e.cause, compat_HTTPError) and e.cause.code == 404:
-                self.raise_no_formats(
-                    'The requested video is not archived, indexed, or there is an issue with web.archive.org',
-                    expected=True)
-            else:
-                raise
+        retry_manager = self.RetryManager(fatal=False)
+        for retry in retry_manager:
+            try:
+                urlh = self._request_webpage(
+                    HEADRequest('https://web.archive.org/web/2oe_/http://wayback-fakeurl.archive.org/yt/%s' % video_id),
+                    video_id, note='Fetching archived video file url', expected_status=True)
+            except ExtractorError as e:
+                retry.error = e
+                # HTTP Error 404 is expected if the video is not saved.
+                if isinstance(e.cause, compat_HTTPError) and e.cause.code == 404:
+                    self.raise_no_formats(
+                        'The requested video is not archived, indexed, or there is an issue with web.archive.org', expected=True)
```

Comment:
```suggestion
                # HTTP Error 404 is expected if the video is not saved.
                if isinstance(e.cause, compat_HTTPError) and e.cause.code == 404:
                    self.raise_no_formats(
                        'The requested video is not archived, indexed, or there is an issue with web.archive.org', expected=True)
                else:
                    retry.error = e
```
So that it doesnt retry when `--ignore-no-format-error`